deepmd/utils/data_system.py

```
@@ -844,14 +1313,483 @@ def prob_sys_size_ext(keywords: str, nsystems: int, nbatch: int) -> list[float]:
+    for pattern in patterns:
+        matches.extend(str(match) for match in root.rglob(pattern))
+    return sorted(set(matches))
+
+
+def _conversion_cache_path(source: Path, fmt: str, out_fmt: str) -> Path:
+    source_resolved = source.resolve(strict=False)
+    try:
+        dpdata_version = importlib.metadata.version("dpdata")
+    except importlib.metadata.PackageNotFoundError:
+        dpdata_version = "unknown"
+    digest = hashlib.sha1(
```

Ruff (0.16.2)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)